tests/metrics/map_mem.sh

```diff
@@ -0,0 +1,287 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
+#
+#  Copyright (C) 2017 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#  Description of the test:
+#  The objective of this tool is to report the memory map of a process in
+#  a JSON file
+
+declare PROC="$1"
+declare NUM_SAMPLES=10
+declare SAMPLES_FILE="rmap.log"
+declare JSON_file
+declare CC_ID
+
+# Checking env
+function init()
+{
+	req_tools=(
+		pmap
+		awk
+		bc)
+
+	if [ -z "$PROC" ];then
+		echo "error: usage: $0 [process name]"
+		exit 1;
+	fi
+
+	# check available tools
+	echo "check required tools"
+	for cmd in ${req_tools[*]};do
+		command -v "$cmd" &> /dev/null
+		res=$?
+		if [ $res -eq 0 ];then
+			echo "command: $cmd: yes"
+			continue
+		fi
+
+		echo  "error: command $cmd not available"
+	done
+
+	# clean env
+	echo "cleaning env"
+	rm -rf *.log
+	rm -rf *.json
```

Comment:
These commands are too loose - the script will delete files that the user might want to keep. It would be better to create a new (empty) temporary directory to store logs in (so that the clean operation isn't required), or only delete named logfiles that this script creates. I think the temporary directory option is better.